azurerm/cache.go

```diff
@@ -9,6 +9,11 @@ import (
 	"github.com/pkg/errors"
 )
 
+// Caches resources, that may be needed for other ones
+// 1 - add cache method
+// 2 - declare method to get the cached resources
+
+//Caching VirtualNetworks
```

Comment:
If you want to add comments then do it the "GO way" which is `// cacheVirtualNetworks ...` and only if you are going to add relevant infomration also hehe, this comment is basically the same as the method name so it's not useful.